src/main/java/essenmakanan/ingredient/IngredientList.java

```diff
@@ -9,29 +10,72 @@ public IngredientList() {
         ingredients = new ArrayList<>();
     }
 
-    public Ingredient getIngredient(int index) {
+    public ArrayList<Ingredient> getIngredients() {
+        return ingredients;
+    }
+
+    public Ingredient getIngredientByIndex(int index) {
         return ingredients.get(index);
     }
 
+    public Ingredient getIngredientByName(String name) {
+        for (Ingredient ingredient : ingredients) {
+            if (ingredient.getName().equals(name)) {
+                return ingredient;
+            }
+        }
+        return null;
+    }
+
+    public int indexOf(Ingredient ingredient) {
+        return ingredients.indexOf(ingredient);
+    }
+
     public void addIngredient(Ingredient ingredient) {
         ingredients.add(ingredient);
     }
 
+    public void editIngredient(Ingredient existingIngredient, Ingredient ingredientToEdit) {
+        String oldName = existingIngredient.getName();
+        String newName = ingredientToEdit.getName();
+
+        String oldQuantity = existingIngredient.getQuantity();
+        String newQuantity = ingredientToEdit.getQuantity();
+
+        IngredientUnit oldUnit = existingIngredient.getUnit();
+        IngredientUnit newUnit = ingredientToEdit.getUnit();
+
+        if (!oldName.equals(newName)) {
+            existingIngredient.setName(newName);
+            Ui.printEditIngredientNameSuccess(oldName, newName);
+        }
+
+        if (!oldQuantity.equals(newQuantity)) {
+            existingIngredient.setQuantity(newQuantity);
+            Ui.printEditIngredientQuantitySuccess(oldQuantity, newQuantity);
+        }
+
+        if (!oldUnit.equals(newUnit)) {
+            existingIngredient.setUnit(newUnit);
+            Ui.printEditIngredientUnitSuccess(oldUnit, newUnit);
+        }
```

Comment:
i think we should implement such that we only change the quantity and the unit of the ingredient, rather than the name also. This is because you check if the name of an ingredient exist in your EditIngredient command.